scripts/build_test.py

```diff
@@ -36,6 +115,211 @@ def test_insert_hash(self):
+        build.ensure_directory_exists(about_staging_filepath)
+        # Create build parent directory (/build/.../pages/about/)
+        build.ensure_directory_exists(about_final_filepath)
+        # Get hash dict for /DEV/.../about folder.
+        file_hashes = build.get_file_hashes(about_source_dir)
+        # Build staging dir for /about (/backend_prod_files/.../about.html).
+        build_tasks = collections.deque()
+        build.build_files(
+            about_source_dir, about_staging_dir, file_hashes, build_tasks,
+            file_formats=('.html,'))
+        # pylint: disable=protected-access
+        build._execute_tasks(build_tasks)
+        # pylint: enable=protected-access
+        #Copy staging about.html to /build/templates/about.[hash].html.
+        copy_tasks = collections.deque()
+        build.copy_files_source_to_target(
+            about_staging_dir, about_build_dir, file_hashes, copy_tasks)
+        # pylint: disable=protected-access
+        build._execute_tasks(copy_tasks)
+        # pylint: enable=protected-access
+
+        # Assert for filepath not containing hash but is supposed to.
+        with self.assertRaises(ValueError) as unhashedFilepath:
+            # pylint: disable=protected-access
+            build._match_directory_with_hashes(
+                about_source_dir, file_hashes)
+        # pylint: enable=protected-access
+        # The first file in the provided dir should be raised as an exception.
+        print unhashedFilepath.exception
+        self.assertTrue(
+            '%s is expected to contain hash' % os.listdir(about_source_dir)[0]
+            in unhashedFilepath.exception)
+
+        # Assert for empty file hash exception.
+        file_hashes.clear()
+        with self.assertRaises(ValueError) as emptyFileHash:
+            # pylint: disable=protected-access
+            build._match_directory_with_hashes(about_staging_dir, file_hashes)
+        # pylint: enable=protected-access
+        print emptyFileHash.exception
+        self.assertTrue('Hash dict is empty' in emptyFileHash.exception)
+
+        # Assert for mismatched hash by using /donate folder's hash dict.
+        donate_source_dir = os.path.join(
+            build.TEMPLATES_DEV_DIR_CORE, 'pages', 'donate')
+        file_hashes.update(build.get_file_hashes(donate_source_dir))
+        with self.assertRaises(KeyError) as incorrectFileHash:
+            # pylint: disable=protected-access
+            build._match_directory_with_hashes(
+                about_build_dir, file_hashes)
+        # pylint: enable=protected-access
+        print incorrectFileHash.exception
+        self.assertTrue(
+            'Hashed file %s does not match hash dict keys'
+            % os.listdir(about_build_dir)[0] in incorrectFileHash.exception)
+        # Clean up directories.
+        shutil.rmtree(about_staging_dir)
+        shutil.rmtree(about_build_dir)
+
+    def test_process_html(self):
```

Comment:
Missing docstring. (It probably isn't needed for test functions, but since you added it to most other tests functions)

Reply:
Done!